src/main/java/uk/gov/hmcts/reform/userprofileapi/controller/advice/UserProfileControllerAdvice.java

```diff
@@ -121,4 +122,25 @@ public static Throwable getRootException(Throwable exception) {
         return new ResponseEntity<>(
                 errorDetails, httpStatus);
     }
+
+    private ResponseEntity<Object> patternErrorDetailsResponseEntity(Exception ex, HttpStatus httpStatus, String errorMsg) {
+        String errorDesc;
+
+        try {
+            errorDesc = ex.getMessage().substring(ex.getMessage().lastIndexOf("default message"));
+            errorDesc = errorDesc.replace("default message [", "").replace("]]", "");
+        } catch (IndexOutOfBoundsException e) {
+            errorDesc = getRootException(ex).getLocalizedMessage();
+        }
+
+        log.error(LOG_STRING, ex);
+        ErrorResponse errorDetails = ErrorResponse.builder()
+                .errorMessage(errorMsg)
+                .errorDescription(errorDesc)
+                .timeStamp(getTimeStamp())
+                .build();
+
+        return new ResponseEntity<>(
+                errorDetails, httpStatus);
+    }
```

Comment:
This method will extract only the default message we have provided. Without this method we receive a large default message generated by [Pattern](https://docs.oracle.com/javaee/7/api/javax/validation/constraints/Pattern.html). However, if no _default message_ is present in the String, it will revert back to how we previously extract the error message.

_Note: this will ONLY be used for the MethodArgumentNotValidException exception_